app/src/ereferrals/java/org/eyeseetea/malariacare/data/database/datasources/CSVVersionLocalDataSource.java

```diff
@@ -0,0 +1,71 @@
+package org.eyeseetea.malariacare.data.database.datasources;
+
+import android.content.Context;
+import android.util.Log;
+
+import com.opencsv.CSVReader;
+
+import org.eyeseetea.malariacare.data.IDataSourceCallback;
+import org.eyeseetea.malariacare.data.database.utils.PopulateDBStrategy;
+import org.eyeseetea.malariacare.data.database.utils.populatedb.PopulateDB;
+import org.eyeseetea.malariacare.domain.boundary.repositories.ICSVVersionRepository;
+
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+
+public class CSVVersionLocalDataSource implements ICSVVersionRepository {
```

Comment:
Repositories and Datasources have the goal to the  domain abstraction of data layer and get and save domain entities but download csv process is not invoked from domain and is not entities, then DataSources and repositories concepts are wrong in this process, create interfaces for abstract is not necessary in this case because generate an unnecessary noise and complexity.